zstash/extract.py

```diff
@@ -162,14 +189,30 @@ def extract_database(
     logger.debug("Max size  : {}".format(config.maxsize))
     logger.debug("Keep local tar files  : {}".format(config.keep))
 
-    # Find matching files
     matches_: List[TupleFilesRow] = []
-    for args_file in args.files:
-        cur.execute(
-            u"select * from files where name GLOB ? or tar GLOB ?",
-            (args_file, args_file),
+    if args.tars is not None:
+        tar_names_initial: List[Tuple[str]] = cur.execute(
+            u"select distinct tar from files"
+        ).fetchall()
+        tar_names: List[str] = sorted([x for (x,) in tar_names_initial])
+        # Remove `.tar` with `[:-4]` for `parse_tars_option` to work properly
+        tar_list: List[str] = parse_tars_option(
+            args.tars, tar_names[0][:-4], tar_names[-1][:-4]
         )
-        matches_ = matches_ + cur.fetchall()
+        for tar in tar_list:
+            cur.execute(
+                u"select * from files where tar GLOB ?",
+                (tar + ".tar",),
+            )
+            matches_ = matches_ + cur.fetchall()
+    else:  # TODO: should this always be run (i.e., not if/else?)
+        # Find matching files
+        for args_file in args.files:
+            cur.execute(
+                u"select * from files where name GLOB ? or tar GLOB ?",
+                (args_file, args_file),
+            )
+            matches_ = matches_ + cur.fetchall()
```

Comment:
Should we still run this section if `--tars` is being used? That is, is there a use case where a user would specify `zstash check --hpss=<HPSS> --tars=<tars> <specific files>`? It seems like if a user is requesting specific files, it wouldn't be necessary to specify which tars to look at.

Reply:
That's right. When specifying the tar files, we don't need the `--tars` option - It's redundant.

Reply:
@tangq If a user specifies both `<tars>` and `<specific files>`, what should be the expected behavior?
1. Just check/extract `<tars>`
2. Just check/extract `<specific files>`
3. Print error message saying to only specify one

Reply:
I'd say option 3. What do you think?

Reply:
I would agree. @golaz which is your preference?

Reply:
@forsyth2 and @tangq : this is an edge case and I would think that it is unlikely a user would want to specify `<tars>` and `<specific files>` simultaneously. Therefore, if it's relatively straightforward to implement, I would suggest option 3 as well to make sure there is no room for misunderstanding from an unsuspecting user.